### ISSUE TITLE
Assertion failure: this->is<js::TypedArrayObject>() with bufferData

### ISSUE THREAD

**Issue by Ms2ger** [servo/servo#6601]
We should pay more attention to JSAPI usage…
```html
<!DOCTYPE html>
<canvas></canvas>
<script>
var c = document.querySelector("canvas");
var ctx = c.getContext("webgl");
var buffer = new ArrayBuffer(8);
var dv = new DataView(buffer);
ctx.bufferData(0, dv, 0);
</script>
```

**Comment by pcwalton** [comment]
Yikes. https://github.com/servo/servo/blob/master/components/script/dom/webglrenderingcontext.rs#L261

Was it necessary to drop down to unsafe code here? Couldn't we have written safe wrappers?

**Comment by jdm** [comment]
This code was written before the SpiderMonkey upgrade, when there was very little support for typed arrays. We can now improve this state of affairs.

**Comment by pcwalton** [comment]
Sweet :)

**Comment by KiChjang** [comment]
Is this fixed by #9065?

**Comment by Ms2ger** [comment]
Seems fixed.
